src/ldp/http/ConvertingErrorHandler.ts

```diff
@@ -0,0 +1,88 @@
+import type {
+  RepresentationConverter,
+  RepresentationConverterArgs,
+} from '../../storage/conversion/RepresentationConverter';
+import { INTERNAL_ERROR } from '../../util/ContentTypes';
+import { getStatusCode } from '../../util/errors/ErrorUtil';
+import { toLiteral } from '../../util/TermUtil';
+import { HTTP, XSD } from '../../util/Vocabularies';
+import { BasicRepresentation } from '../representation/BasicRepresentation';
+import type { Representation } from '../representation/Representation';
+import { RepresentationMetadata } from '../representation/RepresentationMetadata';
+import type { ErrorHandlerArgs } from './ErrorHandler';
+import { ErrorHandler } from './ErrorHandler';
+import type { ResponseDescription } from './response/ResponseDescription';
+
+// Used by internal helper function
+type PreparedArguments = {
+  statusCode: number;
+  conversionArgs: RepresentationConverterArgs;
+};
+
+/**
+ * Converts an error into a Representation of content type internal/error.
+ * Then feeds that representation into its converter to create a representation based on the given preferences.
```

Comment:
Why isn't this just a convertor though, that can be chained?
(I think the answer is: because we'd have to chain it every time.)

But perhaps default the `convertor` argument to a no-op convertor, so it can be omitted?

Reply:
The reason is because we need a class somewhere inbetween to handle the type change (which is this class). Converters take as input a Representation (+ some extra) and return a Representation. But what we have is an Error, and what we want is a ResponseDescription (because that is what the ResponseWriter takes). For succesful requests the OperationHandlers handle the Representation -> ResponseDescription conversion, and the conversion happens somewhere in the store chain. Since errors can also occur outside of those classes, we want to handle this type change as close to the root as possible.

Reply:
Perfect, thanks.